server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/poloniex/PoloniexExchange.java

```diff
@@ -0,0 +1,104 @@
+import org.knowm.xchange.dto.account.AccountInfo;
+import org.knowm.xchange.dto.account.Wallet;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class PoloniexExchange extends XChangeExchange {
+
+    public PoloniexExchange(String preferredFiatCurrency) {
+        super(getDefaultSpecification(), preferredFiatCurrency);
+    }
+
+    public PoloniexExchange(String key, String secret, String preferredFiatCurrency) {
+        super(getSpecification(key, secret), preferredFiatCurrency);
+    }
+
+    private static ExchangeSpecification getDefaultSpecification() {
+        return new org.knowm.xchange.poloniex.PoloniexExchange().getDefaultExchangeSpecification();
+    }
+
+    private static ExchangeSpecification getSpecification(String key, String secret) {
+        ExchangeSpecification spec = getDefaultSpecification();
+        spec.setApiKey(key);
+        spec.setSecretKey(secret);
+        return spec;
+    }
+
+    @Override
+    public Set<String> getCryptoCurrencies() {
+        Set<String> cryptoCurrencies = new HashSet<>();
+        cryptoCurrencies.add(CryptoCurrency.BCH.getCode());
+        cryptoCurrencies.add(CryptoCurrency.BTC.getCode());
+        cryptoCurrencies.add(CryptoCurrency.BURST.getCode());
+        cryptoCurrencies.add(CryptoCurrency.ETH.getCode());
+        cryptoCurrencies.add(CryptoCurrency.LTC.getCode());
+        cryptoCurrencies.add(CryptoCurrency.DASH.getCode());
+        cryptoCurrencies.add(CryptoCurrency.DAI.getCode());
+        cryptoCurrencies.add(CryptoCurrency.DGB.getCode());
+        cryptoCurrencies.add(CryptoCurrency.DOGE.getCode());
+        cryptoCurrencies.add(CryptoCurrency.LSK.getCode());
+        cryptoCurrencies.add(CryptoCurrency.NXT.getCode());
+        cryptoCurrencies.add(CryptoCurrency.POT.getCode());
+        cryptoCurrencies.add(CryptoCurrency.REP.getCode());
+        cryptoCurrencies.add(CryptoCurrency.SYS.getCode());
+        cryptoCurrencies.add(CryptoCurrency.VIA.getCode());
+        cryptoCurrencies.add(CryptoCurrency.XMR.getCode());
+        cryptoCurrencies.add(CryptoCurrency.XRP.getCode());
+        cryptoCurrencies.add(CryptoCurrency.XPM.getCode());
+        return cryptoCurrencies;
+    }
+
+    @Override
+    public Set<String> getFiatCurrencies() {
+        Set<String> fiatCurrencies = new HashSet<>();
+        fiatCurrencies.add(FiatCurrency.USD.getCode());
+        return fiatCurrencies;
+    }
+
+    @Override
+    protected boolean isWithdrawSuccessful(String result) {
+        return true;
+    }
+
+    @Override
+    protected double getAllowedCallsPerSecond() {
+        return 10;
+    }
+
+    @Override
+    public Wallet getWallet(AccountInfo accountInfo, String currency) {
+        return accountInfo.getWallet();
+    }
+
+    @Override
+    protected String translateFiatCurrencySymbolToExchangeSpecificSymbol(String currency) {
+        if (FiatCurrency.USD.getCode().equalsIgnoreCase(currency)) {
+            return "USDT";
```

Comment:
Not sure if this is safe assumption, Poloniex has no fiat currencies, right?
We wanted to implement stablecoins on binance, see note in BPUB-933